tests/losses/loss_functions_test.py

```diff
@@ -5,33 +5,67 @@
 
 from asteroid.filterbanks import STFTFB, Encoder, transforms
 from asteroid.losses import PITLossWrapper
-from asteroid.losses import sdr
-from asteroid.losses import singlesrc_mse, pairwise_mse, multisrc_mse
+from asteroid.losses import sdr, mse
 from asteroid.losses import deep_clustering_loss, SingleSrcPMSQE
 from asteroid.losses import SingleSrcNegSTOI
 from asteroid.losses.multi_scale_spectral import SingleSrcMultiScaleSpectral
 
 
+def assert_loss_checks_shape(loss_func, shape, arbitrary_last_dim=False, no_batch_ok=False):
+    """Test that `loss_func` raises a TypeError if you are passing anything that isn't of the expected shape.
+
+    Args:
+        loss_func (callable): The loss to check, signature: `loss_func(x, y) -> Any`
+        shape (tuple): Shape that the loss is expected to accept (without batch dimension).
+        arbitrary_last_dim (bool, optional): Whether the last dimension may be replaced by any number of dimensions.
+        no_batch_ok (bool, optional): Whether no batch dimension is acceptable.
```

Comment:
(Testing suggestions)

```suggestion
        no_batch_ok (bool, optional): Whether having no batch dimension is acceptable.
```